Make Travis test the book

Moxinilian opened the pull request:
This PR makes Travis build all the examples in the book as part of the typical testing process.
This is very fast and will make us avoid the documentation nonsense we had on 0.8 release.

LucioFranco commented:
Looks like all tests are failing properly here, so should be a fix in the book and good to go. Good stuff mox.

Moxinilian commented:
Are you kidding me

Moxinilian commented:
Okay good the git weird stuff is fixed.
I rebased on master to have the book fixed, and added tests to the prefab chapter.
It should now pass the tests properly and be mergeable.

Moxinilian commented:
As I understood what went wrong last time when writing tests for derive macros in the book, I finally managed to add tests to the `#[derive(SystemData)]` part of the book!

jojolepro commented:
Travis still failing

Moxinilian commented:
Yes. However it seems to be caused by a benchmark-as-a-test thing, which is unreliable enough that it might just be a coincidence. I will restart Travis when needed.

torkleyy commented on book/src/prefab.md:
```diff
@@ -1,6 +1,6 @@
 # Prefabs
 
-A `Prefab` in `Amethyst` is at the core a simple list of future entities, where each entry in 
```
Finally, thank you! Either Amethyst or `amethyst`, not `Amethyst`. Biggest issue in the book resolved xD

torkleyy replied:
We should test for this on Travis and make the build fail if somebody wrote `Amethyst` somewhere.

Moxinilian commented:
What is wrong with Travis? It fails linking on all linux builds for no reason at all.
Does anyone here know why?

Moxinilian commented:
Thank you torkleyy for fixing Travis!
bors r+
